### ISSUE TITLE
Cryptic error from MapMaker / make_counts_image

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1609]
I accidentally typed this:
```python
import astropy.units as u
from gammapy.maps import WcsGeom
from gammapy.cube import MapMaker
from gammapy.data import DataStore
data_store = DataStore.from_dir('$GAMMAPY_EXTRA/datasets/cta-1dc/index/gps/')
obs_id = [110380, 111140, 111159]
obs_list = data_store.obs_list(obs_id)
geom = WcsGeom.create(
    skydir=(0, 0),
    npix=(800, 600),
    binsz=0.02,
    coordsys='GAL',
)
maker = MapMaker(geom, offset_max=u.Quantity('2 deg'))
images = maker.run(obs_list)
```
and it blows up with a cryptic error message:
```
$ python temp.py 
|===========================================>--------------------------------------------------------------------------------------|   1 /  3  (33.33%) ETA     0sTraceback (most recent call last):
  File "temp.py", line 15, in <module>
    images = maker.run(obs_list)
  File "/Users/deil/work/code/gammapy/gammapy/cube/new.py", line 324, in run
    self.process_obs(obs)
  File "/Users/deil/work/code/gammapy/gammapy/cube/new.py", line 280, in process_obs
    obs.events, cutout_geom, obs.pointing_radec, self.offset_max,
  File "/Users/deil/work/code/gammapy/gammapy/cube/new.py", line 79, in make_map_counts
    counts_map.data[:, offset_mask] = 0
IndexError: too many indices for array
```

The problem is in `make_map_counts` here:
https://github.com/gammapy/gammapy/blob/a013ff8ac532ab8b15cee95c5da2abb8937bde9c/gammapy/cube/new.py#L79

It doesn't work for 2D images.

There's other obvious issues one encounters when making maps, e.g. replacing `offset_max=u.Quantity('2 deg')` with `offset_max='2 deg'` above gives another cryptic error, because the mapmaker just does `self.offset_max = offset_max` but should do `self.offset_max = Angle(offset_max)` to be kind to users.

The solution is to rewrite the functions in `new.py` to take a mask instead of a max offset, and to improve their test coverage, e.g. also trying to run them on a 2D geom (and either succeed, or error out with a good error message).

I consider this high priority, we should do that tomorrow.

@registerrier - you or me?

**Comment by registerrier** [comment]
Indeed, the code in new.py  expects explicit 3D maps. Having an explicit energy range is necessary for projected irf maps any way. It is probably safe to enforce explicit energy range for the counts map as well. We can add an explicit test for this.
OK for the kind user interface for max_offset. 
I agree that a fov_mask is better than a simple offset cut. I think that it is better to use it to determine the `MapCoord` at which one wants to compute the IRF value rather than computing the value for all positions and then forcing it to 0 outside the valid range defined by the mask.

**Comment by cdeil** [comment]
I'll make a PR for this today.

**Comment by cdeil** [comment]
Fixed in #1622 . Error messages are now good, and I've added a test to establish that MapMaker works if the energy axis only has a single energy bin.
